src/strategy-heat-capacitor-functions.js

```diff
@@ -0,0 +1,181 @@
+    let array_length = buy_prices.length
+    let schedule={start_at: start_date, temperatures: Array(array_length), max_temp_adjustment: max_temp_adjustment, duration_in_minutes: array_length}
+    if(buy_sell_stacked_array[0].length===0){
+        schedule.temperatures.fill(-max_temp_adjustment,0,array_length)
+    } else{
+        n=0
+        for(i=0;i<buy_sell_stacked_array[0].length;i++){
+            schedule.temperatures.fill(-max_temp_adjustment,n,buy_sell_stacked_array[0][i])
+            schedule.temperatures.fill(max_temp_adjustment,buy_sell_stacked_array[0][i],buy_sell_stacked_array[1][i])
+            n=buy_sell_stacked_array[1][i]
+        }
+        schedule.temperatures.fill(-max_temp_adjustment,n,array_length)
+    }
+
+    schedule.trades = calculate_value_dictlist(buy_sell_stacked_array,buy_prices,sell_prices, start_date)
+    return schedule
+}
+
+function find_temp(date, schedule){
+    var diff = Math.round(date.diff(schedule.start_at).as('minutes'))
+    return schedule.temperatures[diff]
+}
+
+function run_buy_sell_algorithm(price_data,time_heat_1c,time_cool_1c,max_temp_adjustment,min_saving_NOK_kWh) {
+    var prices = [...price_data.map((pd) => pd.value)]
+    //dates = [...price_data.map((pd) => pd.start)]
+    var start_date =  DateTime.fromISO(price_data[0].start);
+    //prices= prices.slice(0,16)
+    
+    //pattern for how much power is procured/sold when. 
+    //This has, for now, just a flat aquisition/divestment profile
+    var buy_pattern= Array(Math.round(time_heat_1c*max_temp_adjustment*2)).fill(1)
+    var sell_pattern= Array(Math.round(time_cool_1c*max_temp_adjustment*2)).fill(1)
+    
+    //Calculate what it will cost to procure/sell 1 kWh as a function of time
+    var buy_prices=calculate_opportunities(prices,buy_pattern,1);
+    var sell_prices=calculate_opportunities(prices,sell_pattern,1)
+    
+    //Find dates for when to procure/sell
+    var buy_sell = find_best_buy_sell_pattern(buy_prices,buy_pattern.length,sell_prices,sell_pattern.length)
+        
+    //Remove small/disputable gains (least profitable buy/sell pairs)
+    var buy_sell_cleaned = remove_low_buysell_pairs(buy_sell,buy_prices,sell_prices,min_saving_NOK_kWh, start_date)
+
+    //Calculate temperature adjustment as a function of time
+    let schedule = calculate_schedule(start_date, buy_sell_cleaned, buy_prices, sell_prices, max_temp_adjustment)
+    
+    return schedule
+}
+
+
+function test(){ //Set up price data
+    price_data = decreasing_end_prices.priceData //decreasing_end_prices.priceData;
+    //User input
+    time_heat_1c = 55
+    time_cool_1c = 40
+    max_temp_adjustment = 0.61
+    min_saving_NOK_kWh = 0.1
+    debugger
+    let schedule = run_buy_sell_algorithm(price_data, time_heat_1c, time_cool_1c,max_temp_adjustment,min_saving_NOK_kWh)
+    dT = find_temp(DateTime.local(2022, 1, 2),schedule)
+    debugger
+}
+
+//var value = test()
```

Comment:
This code is not used, so please remove it

Reply:
Done